src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEditPane.ts

```diff
@@ -316,65 +318,76 @@ export class BulkEditPane extends ViewPane {
 		}
 	}
 
-	private async _openElementAsEditor(e: IOpenEvent<BulkEditElement | undefined>): Promise<void> {
+	private async _openElementInMultiDiffEditor(e: IOpenEvent<BulkEditElement | undefined>): Promise<void> {
 
-		const options: Mutable<ITextEditorOptions> = { ...e.editorOptions };
+		const fileOperations = this._currentInput?.fileOperations;
+		if (!fileOperations) {
+			return;
+		}
 		let fileElement: FileElement;
 		if (e.element instanceof TextEditElement) {
 			fileElement = e.element.parent;
-			options.selection = e.element.edit.textEdit.textEdit.range;
-
 		} else if (e.element instanceof FileElement) {
 			fileElement = e.element;
-			options.selection = e.element.edit.textEdits[0]?.textEdit.textEdit.range;
-
 		} else {
 			// invalid event
 			return;
 		}
 
-		const previewUri = this._currentProvider!.asPreviewUri(fileElement.edit.uri);
-
-		if (fileElement.edit.type & BulkFileOperationType.Delete) {
-			// delete -> show single editor
-			this._editorService.openEditor({
-				label: localize('edt.title.del', "{0} (delete, refactor preview)", basename(fileElement.edit.uri)),
-				resource: previewUri,
-				options
-			});
-
-		} else {
-			// rename, create, edits -> show diff editr
-			let leftResource: URI | undefined;
-			try {
-				(await this._textModelService.createModelReference(fileElement.edit.uri)).dispose();
-				leftResource = fileElement.edit.uri;
-			} catch {
-				leftResource = BulkEditPreviewProvider.emptyPreview;
+		const resources = await this._resolveResources(fileOperations);
+		const options: Mutable<ITextEditorOptions> = {
+			...e.editorOptions,
+			viewState: {
+				revealData: {
```

Comment:
Please use an interface here!
When you change `revealData` to `revaelData`, it would still compile!
General rule: When there is a functional dependency, there should be a static code dependency as well.
We have a functional dependency here (`revealData` is set here and read somewhere else), but there is no static dependency (you cannot go to the reader of `revealData` by using go to reference)